### DISCUSSION TITLE
Building-GMP-and-MPFR-on-Windows-with-Visual-Studio

### DISCUSSION THREAD

**Discussion by canberkirimagzi** [microsoft/vcpkg#40649] [Q&A]
Hello. I am getting an Error when I use the command "vcpkg install gmp --triplet=x64-windows" while following the instructions in https://github.com/CGAL/cgal/wiki/Building-GMP-and-MPFR-on-Windows-with-Visual-Studio

My internet connection is quite stable. But I get the same error as below.
How can I resolve this? Thank you.

warning: The vcpkg C:\Program Files\Microsoft Visual Studio\2022\Community\VC\vcpkg\vcpkg.exe is using detected vcpkg root C:\src\vcpkg and ignoring mismatched VCPKG_ROOT environment value C:\Program Files\Microsoft Visual Studio\2022\Community\VC\vcpkg. To suppress this message, unset the environment variable or use the --vcpkg-root command line switch.
error: Could not locate a manifest (vcpkg.json) above the current working directory.
This vcpkg distribution does not have a classic mode instance.

**Comment by autoantwort** [comment]
You must probably call `.\vcpkg install gmp --triplet=x64-windows` to use the vcpkg.exe from the cwd and not the vcpkg.exe from visual studio since it does not support the vcpkg install <package_name> command aka classic mode.

**Reply by canberkirimagzi** [reply]
Thank you brother. "**.\vcpkg** install gmp --triplet=x64-windows" worked (not "./vcpkg" in case someone copies in future). I will not be closing this QA thread until I complete all the steps. Much appreciated.

**Reply by dg0yt** [reply]
Note: Using just `vcpkg` on Windows may call a more limited executable which is bundled with VS components.

**Comment by canberkirimagzi** [comment]
I think I am done with the instructions in https://github.com/CGAL/cgal/wiki/Building-GMP-and-MPFR-on-Windows-with-Visual-Studio
I suppose there is also a linking process of GMP with a project as well that needs to be done. Can anyone provide steps for that as well --though this might not be the appropriate place to ask for it?

This comsumed too much of my time already and can't even tell if I am making a progress or not. I once wrote a code with GMP library and it was not difficult at all compared with the process of reinstalling it now. 

Thank you.

**Reply by Osyotr** [reply]
There are usage files for both GMP and MPFR:
https://github.com/microsoft/vcpkg/blob/bfe8c13a89dfc6d360a9db7fa9d91cb50c58d52b/ports/gmp/usage
https://github.com/microsoft/vcpkg/blob/856e200a1264bf2fcbe7a19b0dcd0ed7aa1cf1bd/ports/mpfr/usage

**Reply by canberkirimagzi** [reply]
The package gmp can be imported via CMake FindPkgConfig module:

    #  gmp
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmp REQUIRED IMPORTED_TARGET gmp)
    target_link_libraries(main PkgConfig::gmp)

    # gmpxx
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(gmpxx REQUIRED IMPORTED_TARGET gmpxx)
    target_link_libraries(main PkgConfig::gmpxx)



I appreciate. I am afraid I don't really know what to do with this. ChatGPT gave directions but didn't work. I don't have a proper terminology even to chase the folders
